pkg/metrics/listener.go

```diff
@@ -41,8 +44,39 @@ func NewListener(addr string) (net.Listener, error) {
 		return nil, nil
 	}
 
-	log.Info("Metrics server is starting to listen", "addr", addr)
-	ln, err := net.Listen("tcp", addr)
+	if !secureServing {
+		log.Info("Metrics server is starting to listen", "addr", addr)
+		ln, err := net.Listen("tcp", addr)
+		if err != nil {
+			er := fmt.Errorf("error listening on %s: %w", addr, err)
+			log.Error(er, "metrics server failed to listen. You may want to disable the metrics server or use another port if it is due to conflicts")
+			return nil, er
+		}
+		return ln, nil
+	}
+
+	// Note: Using self-signed certificates here should be good enough as usually nobody goes through
```

Comment:
I almost guarantee you that immediately someone will ask for a way to submit their own certs 🙃 Openshift for example has a controller that generates them based on some service annotations, cert-manager can do the same...Maybe provide functional opts similiar as we do for the webhook endpoints (`...func(*tls.Config)`) and only generate these if Certificates is unset?

Reply:
Should we consider to offer a way to use the same certificates as the webhooks?

Reply:
Should both work now